server/image.go

```diff
@@ -189,9 +198,55 @@ func (im *Image) Release() {
 	}
 }
 
-func (im *Image) sync() {
+func (im *Image) sync() error {
 	im.Width = im.img.Width()
 	im.Height = im.img.Height()
+
+	if im.Format == "mp4" && len(im.Data) != 0 {
+		// No need to convert again.
+		return nil
+	}
+
 	im.Format = im.img.Format()
 	im.Data = im.img.Data()
+
+	if im.conversionFormat == "mp4" && im.Format == "gif" {
```

Comment:
ah. well.. lets just write to disk like you’re doing  


On Wednesday, December 16, 2015 at 10:28 AM, José Carlos wrote:

> In server/image.go (https://github.com/pressly/imgry/pull/30#discussion_r47789294):
> > im.Format = im.img.Format() > im.Data = im.img.Data() > + > + if im.conversionFormat == "mp4" && im.Format == "gif" {  
> I was thinking of doing something like that, but the problem is that ffmpeg does not recognize the input stream, according to the docs, this should work:
> cat issue-8.500.gif | ffmpeg -f image2pipe -i - -y foo.mp4  
> However it throws:
> ... [image2pipe @ 0x2fa33e0] Could not find codec parameters for stream 0 (Video: none, none): unknown codec Consider increasing the value for the 'analyzeduration' and 'probesize' options pipe:: could not find codec parameters ...  
> I've found other people with similar problems but without an answer, like https://ffmpeg.org/pipermail/ffmpeg-user/2012-March/005677.html
>  
> —
> Reply to this email directly or view it on GitHub (https://github.com/pressly/imgry/pull/30/files#r47789294).
>  
>  
>  
>  
>